src/libsync/accessmanager.cpp

```diff
@@ -78,9 +78,41 @@ QNetworkReply *AccessManager::createRequest(QNetworkAccessManager::Operation op,
         newRequest.setAttribute(QNetworkRequest::Http2AllowedAttribute, http2EnabledEnv);
     }
 
+    auto sslConfiguration = newRequest.sslConfiguration();
+
+    // for some reason, passing an empty list causes the default chain to be removed
+    // this behavior does not match the documentation
+    if (!_customTrustedCaCertificates.isEmpty()) {
+        sslConfiguration.addCaCertificates({ _customTrustedCaCertificates.begin(), _customTrustedCaCertificates.end() });
+    }
+
+    newRequest.setSslConfiguration(sslConfiguration);
+
     const auto reply = QNetworkAccessManager::createRequest(op, newRequest, outgoingData);
     HttpLogger::logRequest(reply, op, outgoingData);
     return reply;
 }
 
+QSet<QSslCertificate> AccessManager::customTrustedCaCertificates()
+{
+    return _customTrustedCaCertificates;
+}
+
+void AccessManager::setCustomTrustedCaCertificates(const QSet<QSslCertificate> &certificates)
+{
+    _customTrustedCaCertificates = certificates;
+}
+
+void AccessManager::addCustomTrustedCaCertificates(const QList<QSslCertificate> &certificates)
+{
+    std::for_each(certificates.begin(), certificates.end(), [this](const QSslCertificate &certificate) {
```

Comment:
{ _customTrustedCaCertificates.begin(), _customTrustedCaCertificates.end() } ?

Reply:
I'm not sure what you mean.

(I tried `std::copy`, but the Qt containers don't work with STL algorithms.)